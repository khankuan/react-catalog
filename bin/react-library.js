#!/usr/bin/env node
var spawn = require('cross-spawn')
var result = spawn.sync(
  'babel-node',
  [require.resolve('../dist/index.js')].concat(process.argv.slice(2)),
  {stdio: 'inherit'}
)
process.exit(result.status)
