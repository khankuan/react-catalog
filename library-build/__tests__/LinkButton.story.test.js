
import React from 'react'
import renderer from 'react-test-renderer'
import story from '../../docs/src/LinkButton/LinkButton.story.js'
import { Story } from 'react-library'

jest.useFakeTimers()

story.stories.forEach(s => {
  if (s.sequence) {
    const Component = story.type
    let props = Story.getSequenceInitialProps(s)
    let ref
    const node = renderer.create(<Component {...props} ref={_ref => (ref = _ref)} />)
    s.sequence.forEach(async function(step, i) {
      it(`LinkButton.story - ${s.title} - ${step.title}`, async () => {
        if (i > 0) {
          const result = Story.getSequenceStepProps(ref, s, i, props)
          jest.runAllTimers()
          props = await result
          node.update(<Component {...props} ref={_ref => (ref = _ref)} />)
        }
        const tree = node.toJSON()
        expect(tree).toMatchSnapshot()
        expect(story.popHandleCalls()).toMatchSnapshot()
      })
    })
  } else {
    let node
    if (React.isValidElement(s.content)) {
      node = s.content
    } else {
      const Component = story.type
      node = <Component {...s.content} />
    }
    it(`LinkButton.story - ${s.title}`, () => {
      expect(renderer.create(node)).toMatchSnapshot()
    })
  }
})
