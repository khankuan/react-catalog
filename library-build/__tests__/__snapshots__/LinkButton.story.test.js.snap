exports[`test LinkButton.story - Default 1`] = `
<a
  className="demo-link-button">
  Click me
</a>
`;
